components/frontend/src/header_footer/buttons/ReportPeriodButton.jsx

```diff
@@ -0,0 +1,30 @@
+import ArrowDropDownIcon from "@mui/icons-material/ArrowDropDown"
+import ArrowRightIcon from "@mui/icons-material/ArrowRight"
+import EventIcon from "@mui/icons-material/Event"
+import { Stack } from "@mui/material"
+import { bool, func } from "prop-types"
+
+import { datePropType } from "../../sharedPropTypes"
+import { AppBarButton } from "./AppBarbutton"
+
+export function ReportPeriodButton({ reportDate, setSettingsPanelVisible, settingsPanelVisible }) {
+    return (
+        <AppBarButton
+            tooltip="Change report date, number of dates shown, time between dates, and date order"
+            startIcon={
+                <Stack direction="row" spacing={0}>
+                    {settingsPanelVisible ? <ArrowDropDownIcon /> : <ArrowRightIcon />}
+                    <EventIcon />
+                </Stack>
+            }
+            onClick={() => setSettingsPanelVisible(!settingsPanelVisible)}
+        >
+            {reportDate ? reportDate.toLocaleDateString() : "today"}
+        </AppBarButton>
+    )
+}
+ReportPeriodButton.propTypes = {
+    reportDate: datePropType,
+    setSettingsPanelVisible: func,
+    settingsPanelVisible: bool,
```

Comment:
Wasn't this now the string `visibleSettingsPanel`?

Reply:
No, see above. The buttons only need to know whether their own panel is visible or not (hence a boolean). The menubar keeps track of which panel is visible, hence a string there.